src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx

```diff
@@ -1,63 +1,161 @@
+      if (details) {
+        const placeDetails = await getPlaceDetails(details.place_id);
+
+        if (!placeDetails) {
+          showAlert('error', { message: 'Something went wrong!' });
+          return;
+        }
+        const formattedLocation = formatLocationString(placeDetails.formatted_address);
+        setLocation(formattedLocation)
+      }
+    },
+    [getPlaceDetails, formatLocationString, setLocation]
+  );
+
+  useEffect(() => {
+    setFormValues({ ...filter, location, checkInDate, checkOutDate });
+  }, [filter, location, checkInDate, checkOutDate]);
+
+  const handleCheckInChange = useCallback(
+    (newDate: string) => {
+      setcheckInDate(newDate);
+    },
+    [setcheckInDate]
+  );
+
+  const handleCheckOutChange = useCallback(
+    (newDate: string) => {
+      setcheckOutDate(newDate);
+    },
+    [setcheckOutDate]
+  );
+
+  const getCheckOutMinDate = useCallback(() => {
+    const checkoutMin = checkInDate ? getNextDay(checkInDate) : getNextDay(getToday());
+    return checkoutMin;
+  }, [checkInDate, getNextDay, getToday]);
+
+  const getCheckInMinDate = useCallback(() => {
+    const today = getToday();
+    return today;
+  }, [getToday]);
+
 
-  const [isCollapsed, setIsCollapsed] = useState(COLLAPSABLE_CARDS_POSITIONS.wherePressed);
+  const clearCheckin = useCallback(() => {
+      setcheckInDate('');
+    }, [setcheckInDate]);
+  
+  const clearCheckout = useCallback(() => {
+    setcheckOutDate('');
+  },[setcheckOutDate])
 
-  const toggleCollapseWhen = () => {
-    setIsCollapsed(COLLAPSABLE_CARDS_POSITIONS.whenPressed);
+  const clearLocation = useCallback(() => {
+    setLocation('');
+  }, [setLocation]);
+
+  const handleSearch = useCallback(() => {
+    if (isInvalidDateRange(checkInDate, checkOutDate)) {
+      showAlert('error', { message: 'Date range is invalid' });
+      return;
+    }
+    dispatch(accommodationListActions.setFilter(formValues));
+    changeOpen();
+    return;
+  }, [isInvalidDateRange, dispatch, changeOpen, formValues]);
+
+  const handleResetSearch = useCallback(() => {
+    setLocation('');
+    setcheckInDate('');
+    setcheckOutDate('');
+  }, [setLocation,setcheckOutDate, setcheckInDate]);
+
+  const [isCollapsed, setIsCollapsed] = useState(COLLAPSABLE_CARDS_POSITIONS.allClosed);
+
+  const toggleCollapseLocation = () => {
+    isCollapsed.location ? 
+      setIsCollapsed(COLLAPSABLE_CARDS_POSITIONS.locationPressed) :
+      setIsCollapsed(COLLAPSABLE_CARDS_POSITIONS.allClosed);
   };
 
-  const toggleCollapseWhere = () => {
-    setIsCollapsed(COLLAPSABLE_CARDS_POSITIONS.wherePressed);
+  const toggleCollapseCheckIn = () => {
+    isCollapsed.checkin ? 
+    setIsCollapsed(COLLAPSABLE_CARDS_POSITIONS.checkInPressed) :
+    setIsCollapsed(COLLAPSABLE_CARDS_POSITIONS.allClosed);
   };
+
+  const toggleCollapseCheckOut = () => {
+    isCollapsed.checkout ? 
+    setIsCollapsed(COLLAPSABLE_CARDS_POSITIONS.checkOutPressed) :
```

Comment:
if you need to change related values several times inside a component, better to use useReducer, not mandatory, just keep in mind

Reply:
sure, moved toggling all collapsable sections from useState to useReducer